client/web/compose/src/components/PageBlocks/RecordListBase.vue

```diff
@@ -1110,9 +1115,39 @@ export default {
 
     onUpdateFields (fields = []) {
       this.options.fields = [...fields]
+      this.setStorageRecordListConfiguredFields()
+
       this.$emit('save-fields', this.options.fields)
     },
 
+    setStorageRecordListConfiguredFields () {
+      let currentConfiguredFields = []
+
+      try {
+        // Get record list configured fields from localStorage
+        currentConfiguredFields = getItem(`record-list-configured-columns-${this.uniqueID}`)
+        currentConfiguredFields = this.options.fields.map((f) => f.fieldID)
+
+        setItem(`record-list-configured-columns-${this.uniqueID}`, currentConfiguredFields)
+      } catch (e) {
+        console.warn(this.$t('notification:record-list.corrupted-configured-fields'))
+      }
+    },
+
+    getStorageRecordListConfiguredFields () {
+      try {
+        // Get record list configured fields from localStorage
+        const currentConfiguredFields = getItem(`record-list-configured-columns-${this.uniqueID}`)
+
+        this.customConfiguredFields = currentConfiguredFields
```

Comment:
this.customConfiguredFields = getItem(`record-list-configured-columns-${this.uniqueID}`)